src/dac.rs

```diff
@@ -125,84 +50,7 @@ impl Dac
         });
     }
 
-    pub fn set_trigger(&mut self, channel: DacChannel, trigger: Trigger) {
-        let cr = &self.regs.cr;
-
-        match channel {
-            DacChannel::One => {
-                cr.modify(|_, w| unsafe {
-                    w.ten1().set_bit();
-                    w.tsel1().bits(trigger as u8)
-                });
-            }
-            DacChannel::Two => {
-                cr.modify(|_, w| unsafe {
-                    w.ten2().set_bit();
-                    w.tsel2().bits(trigger as u8)
-                });
-            }
-        }
-    }
-
-    pub fn enable_noise_gen(&mut self, channel: DacChannel, mamp: Mamp) {
-        let cr = &self.regs.cr;
-
-        match channel {
-            DacChannel::One => {
-                cr.modify(|_, w| unsafe {
-                    w.wave1().noise();
-                    w.mamp1().bits(mamp as u8)
-                });
-            }
-            DacChannel::Two => {
-                cr.modify(|_, w| unsafe {
-                    w.wave2().noise();
-                    w.mamp2().bits(mamp as u8)
-                });
-            }
-        }
-
-    }
-
-    pub fn enable_triangle_gen(&mut self, channel: DacChannel, mamp: Mamp) {
-        let cr = &self.regs.cr;
-
-        match channel {
-            DacChannel::One => {
-                cr.modify(|_, w| unsafe {
-                    w.wave1().triangle();
-                    w.mamp1().bits(mamp as u8)
-                });
-            }
-            DacChannel::Two => {
-                cr.modify(|_, w| unsafe {
-                    w.wave2().triangle();
-                    w.mamp2().bits(mamp as u8)
-                });
-            }
-        }
-
-    }
-
     pub fn write_data(&mut self, data: u32) {
-
-        match self.bits {
-            DacBitAlignment::EightRight =>  {
-                self.regs.dhr8r1
-                    .modify(|_,w| unsafe {
-                        w.bits(data)})
-            },
-            DacBitAlignment::TwelveLeft => {
-                self.regs.dhr12l1
-                    .modify(|_,w| unsafe {
-                        w.bits(data)})
-            },
-            DacBitAlignment::TwelveRight => { 
-                self.regs.dhr12r1
-                    .modify(|_,w| unsafe {
-                        w.bits(data)})
-            }
-        }
+        self.regs.dhr8r1.modify(|_, w| unsafe { w.bits(data) })
```

Comment:
the test / example firmware uses 12-bit values so this should use the `dhrl12r1` register or the sine wave will get distorted

it would be great to
- document that `data` parameter MUST be a 12-bit value -- values outside that range will result in misbehavior
- change the type of `data` to `u16`
- use the safe API here: `regs.dhr12r1.write(|w| w.dacc1dhr().bits(data))`

Reply:
fixed